core/common/src/main/java/alluxio/worker/block/BlockWorker.java

```diff
@@ -115,6 +115,26 @@ void commitBlock(long sessionId, long blockId, boolean pinOnCreate)
   String createBlock(long sessionId, long blockId, int tier, String medium, long initialBytes)
       throws BlockAlreadyExistsException, WorkerOutOfSpaceException, IOException;
 
+  /**
+   * Creates a block in Alluxio managed space.
+   * Calls {@link #createBlockWriter} to get a writer for writing to the block.
+   * The block will be temporary until it is committed by {@link #commitBlock} .
+   * Throws an {@link IllegalArgumentException} if the location does not belong to tiered storage.
+   *
+   * @param sessionId the id of the client
+   * @param blockId the id of the block to create
+   * @param tier the tier to place the new block in
+   *        {@link BlockStoreLocation#ANY_TIER} for any tier
+   * @param createBlockOptions the createBlockOptions
+   * @return a string representing the path to the local file
+   * @throws BlockAlreadyExistsException if blockId already exists, either temporary or committed,
+   *         or block in eviction plan already exists
+   * @throws WorkerOutOfSpaceException if this Store has no more space than the initialBlockSize
+   */
+  String createBlock(long sessionId, long blockId, int tier,
```

Comment:
@apc999  changed accordingly, please tell me if there are more change required, so that I can change. After this change I will change the EE side accordingly

Reply:
@yyongycy prefer to keep all parameters in the createBlockOptions for cleanness, let me know if you have other opinions

Reply:
especially when medium and tier are really relavent